test/some.jl

```diff
@@ -75,6 +77,10 @@ for v in (nothing, missing)
     @test coalesce(v, nothing) === nothing
     @test coalesce(v, missing, v) === v
     @test coalesce(v, nothing, v) === v
+
+    # issue #26927
+    @test all(coalesce.([missing, nothing, Some(nothing), Some(missing)], "replacement") .===
```

Comment:
Actually `coalesce` isn't involved here: the error happens without it, when constructing the array. Better drop it since it obscures what is being tested.